Src/BootCamp.Chapter/Csv/CsvBase.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter.Csv
+{
+    public class CsvBase
+    {
+        public CsvRow Header { get; protected set; } = new CsvRow();
+        public string Footer { get; protected set; } = string.Empty;
+        public CsvDelimiter Delimiter { get; protected set; } = CsvDelimiter.Comma;
+        protected string FileName { get; set; }
+        protected bool HasHeader { get; set; }
+        protected bool HasFooter { get; set; }
+        protected IList<CsvRow> Rows { get; set; } = new List<CsvRow>();
+
+        public CsvBase(string fileName)
+        {
+            if (!fileName.IsValid())
```

Comment:
Overkill to have an extension method for validating a string like this. It's not much different from string.IsNullOrEmpty, however what is a valid fileName I would need to dig. In fact, I would need to go as far as extensions.

Extension methods are used for giving extra functionality to existing classes. They are not supposed to contain business logic, just that data structure-specific logic. IsValid is logic that belong to CsvBase and not string.

Reply:
IsValid just checks the string if it's null, white space or empty

Reply:
yup, and that is alraedy done for you- string.IsNullOrWhitespace

Reply:
I think I've been using IsValid as a stanalone method since Chapter-1 and I would have swore that it does not check for empty string. I allways used both IsNullOrWhitespace || IsNullOrEmpty

Oh well it better late than never 😄